src/steamship/experimental/transports/telegram.py

```diff
@@ -0,0 +1,111 @@
+from steamship.experimental.transports.chat import ChatMessage
+from steamship.experimental.transports.transport import Transport
+
+API_BASE = "https://api.telegram.org/bot"
+
+
+class TelegramTransport(Transport):
+    """Experimental base class to encapsulate a Telegram communication channel."""
+
+    api_root: str
+
+    def __init__(self, bot_token: str):
+        self.api_root = f"{API_BASE}{bot_token}"
+
+    def _instance_init(self, *args, **kwargs):
+        if "webhook_url" not in kwargs:
+            raise SteamshipError(
+                message="Please provide the `webhook_url` method in the TelegramTransport instance_init kwargs"
+            )
+
+        webhook_url = kwargs["webhook_url"]
+        logging.info(f"Setting Telegram webhook URL: {webhook_url}")
+        logging.info(f"Post is to {self.api_root}/setWebhook")
+
+        response = requests.get(
+            f"{self.api_root}/setWebhook",
+            params={
+                "url": webhook_url,
+                "allowed_updates": ["message"],
+                "drop_pending_updates": True,
+            },
+        )
+
+        if not response.ok:
+            raise SteamshipError(
+                f"Could not set webhook for bot. Webhook URL was {webhook_url}. Telegram response message: {response.text}"
+            )
+
+    def _instance_deinit(self, *args, **kwargs):
+        """Unsubscribe from Telegram updates."""
+        requests.get(f"{self.api_root}/deleteWebhook")
+
+    def _send(self, blocks: [ChatMessage]):
+        """Send a response to the Telegram chat."""
+        for block in blocks:
+            chat_id = block.get_chat_id()
+            if block.is_text() or block.text:
+                params = {"chat_id": int(chat_id), "text": block.text}
+                requests.get(f"{self.api_root}/sendMessage", params=params)
+            elif block.is_image():
+                _bytes = block.raw()
+                with tempfile.TemporaryFile(mode="r+b") as temp_file:
+                    temp_file.write(_bytes)
+                    temp_file.seek(0)
+                    resp = requests.post(
+                        url=f"{self.api_root}/sendPhoto?chat_id={chat_id}",
+                        files={"photo": temp_file},
+                    )
+                    if resp.status_code != 200:
+                        logging.error(f"Error sending message: {resp.text} [{resp.status_code}]")
+                        raise SteamshipError(f"Message not sent successfully: {resp.text}")
+            else:
+                logging.error(
+                    f"Telegram transport unable to send a block of MimeType {block.mime_type}"
+                )
+
+    def _info(self) -> dict:
+        """Fetches info about this bot."""
+        resp = requests.get(f"{self.api_root}/getMe").json()
+        logging.info(f"/info: {resp}")
+        return {"telegram": resp.get("result")}
+
+    def _parse_inbound(self, payload: dict, context: Optional[dict] = None) -> ChatMessage:
```

Comment:
should we have an option for `update_id` and caching the last N? of them so that we avoid duplications?

Reply:
100%. I think the next round of work on this could start to work with the `ChatFile` class that @dkolas has been working on to make sure we don't duplicate inbound or outbound.

Does the LLM caching work you've been doing related to this? Or is that a separate set of code?